app/src/behaviors/behavior_macro.c

```diff
@@ -209,9 +213,98 @@ static int on_macro_binding_released(struct zmk_behavior_binding *binding,
     return ZMK_BEHAVIOR_OPAQUE;
 }
 
+#if IS_ENABLED(CONFIG_ZMK_BEHAVIOR_METADATA)
+
+static int get_macro_parameter_metadata(const struct device *macro,
```

Comment:
This function is massive. Any way it could be broken into smaller pieces?

The combination of loop conditions with param1_found/param2_found and continue/break statements is particularly difficult to understand.

Reply:
Ok, cleaned this up a ton, after properly identifying the edges. Read better?

Reply:
Might warrant some extra comments explaining exactly what this is doing, but I *think* I correctly understand what it's doing now, so definitely an improvement.

Just to be sure, this is looping through the list of bindings in the macro, looking for the first thing it can find that uses each of the macro parameters, then returning a combination of the sets for each of those? I'm not sure I understand why the logic for param1 is different than param1 though. Does the param1 case not need to check if the bound param2 is valid?